examples_tests/ComputePointCloud/main.cpp

```diff
@@ -0,0 +1,516 @@
+		auto metadataPly = cpuBundlePLYData.second->selfCast<const asset::CPLYMetadata>();
+
+		// Get the positions from the mesh
+		uint32_t positionVbOffset;
+		core::smart_refctd_ptr<video::IGPUBuffer> positionVb;
+
+		{
+			const asset::ICPUMeshBuffer* const firstMeshBuffer = cpuMeshPly->getMeshBuffers().begin()[0];
+			const asset::SBufferBinding<const asset::ICPUBuffer> posBufferBinding = firstMeshBuffer->getVertexBufferBindings()[firstMeshBuffer->getPositionAttributeIx()];
+			core::smart_refctd_ptr<const asset::ICPUBuffer> posBuffer = posBufferBinding.buffer;
+
+			//const float* positions = static_cast<const float*>(posBuffer->getPointer());
+			//logger->log("Mesh buffer %i bytes %i", system::ILogger::E_LOG_LEVEL::ELL_INFO, 0, posBuffer->getSize());
+			//for (uint32_t i = 0; i < posBuffer->getSize() / 12; i++) {
+			//	logger->log("Pos %f %f %f", system::ILogger::E_LOG_LEVEL::ELL_INFO, positions[i * 3], positions[i * 3 + 1], positions[i * 3 + 2]);
+			//}
+
+			m_pointCount = posBuffer->getSize() / 12;
+			auto posBuffer_cpu = posBuffer.get();
+			auto positionsVertexBuffer = cpu2gpu.getGPUObjectsFromAssets(&posBuffer_cpu, &posBuffer_cpu + 1, cpu2gpuParams)->front();
+			positionVbOffset = positionsVertexBuffer->getOffset();
+			positionVb = core::smart_refctd_ptr<video::IGPUBuffer>(positionsVertexBuffer->getBuffer());
+		}
+
+		// Fill out the descriptor sets
+		{
+			const uint32_t writeDescriptorCount = 2u;
+
+			video::IGPUDescriptorSet::SDescriptorInfo descriptorInfos[writeDescriptorCount];
+			video::IGPUDescriptorSet::SWriteDescriptorSet writeDescriptorSets[writeDescriptorCount] = {};
+
+			// Visbuffer
+			{
+				descriptorInfos[0].image.imageLayout = asset::EIL_GENERAL;
+				descriptorInfos[0].image.sampler = nullptr;
+				descriptorInfos[0].desc = m_visbufferView;
+
+				writeDescriptorSets[0].dstSet = m_rasterizeDescriptorSet.get();
+				writeDescriptorSets[0].binding = 0u;
+				writeDescriptorSets[0].arrayElement = 0u;
+				writeDescriptorSets[0].count = 1u;
+				writeDescriptorSets[0].descriptorType = asset::EDT_STORAGE_IMAGE;
+				writeDescriptorSets[0].info = &descriptorInfos[0];
+			}
+
+			// Point cloud vertex buffer
+			{
+				descriptorInfos[1].buffer.offset = positionVbOffset;
+				descriptorInfos[1].buffer.size = positionVb->getSize();
+				descriptorInfos[1].desc = positionVb;
+
+				writeDescriptorSets[1].dstSet = m_rasterizeDescriptorSet.get();
+				writeDescriptorSets[1].binding = 1u;
+				writeDescriptorSets[1].arrayElement = 0u;
+				writeDescriptorSets[1].count = 1u;
+				writeDescriptorSets[1].descriptorType = asset::EDT_STORAGE_BUFFER;
+				writeDescriptorSets[1].info = &descriptorInfos[1];
+			}
+
+			logicalDevice->updateDescriptorSets(writeDescriptorCount, writeDescriptorSets, 0u, nullptr);
+		}
+
+		{
+			asset::SPushConstantRange pcRange = {};
+			pcRange.stageFlags = asset::IShader::ESS_COMPUTE;
+			pcRange.offset = 0u;
+			pcRange.size = (2 + 1 + 16) * sizeof(uint32_t);
```

Comment:
I'd either make a shared "header" between c++ and GLSL, or get this value from shader introspector